server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -316,11 +315,14 @@ public ValuesSource getField(FieldContext fieldContext, AggregationScript.LeafFa
                         context.query().visit(new QueryVisitor() {
                             @Override
                             public QueryVisitor getSubVisitor(BooleanClause.Occur occur, Query parent) {
-                                // Ignore queries that aren't "MUST"
-                                if (occur != Occur.MUST) {
-                                    return QueryVisitor.EMPTY_VISITOR;
+                                // Only queries that must be included
```

Comment:
I think this comment could be clearer.  On first read, I took it to mean "this switch only lists the queries it has to", not "the switch matches queries who's conditions are not optional"